shared/health/src/main/java/io/pravega/shared/health/HealthEndpoint.java

```diff
@@ -112,4 +112,5 @@ default boolean isAlive() {
      * @return The {@link Health} of the whole health service.
      */
     Health getHealth();
+
```

Comment:
I'm not terribly familiar with how the operators work or what information is has available to them. From my perspective the only thing to mention is the use of the liveness/readiness routes.

Reply:
Operators are first hand users of this Health Framework, so it it may help to understand their needs. Today, they only invoke Readiness and Liveness checks on a component (top level component - controller/segment store & Bookkeeper) and expect to get a boolean result indicating weather the check passed or failed. It may help to update the PDP with details of how these can be invoked for each component.

Reply:
How would the Bookkeeper operator check on BK Health? Is it expected to be a sub-component inside Segment Store, since there is no separate Health Service for Bookkeeper...

Reply:
> 
> 
> How would the Bookkeeper operator check on BK Health? Is it expected to be a sub-component inside Segment Store, since there is no separate Health Service for Bookkeeper...

A `HealthContributor` would need to be implemented for this purpose, which would then be called via the `/v1/health/readiness/bookkeeper` route.

Reply:
Ofcourse a HealthContributor needs to be implemented... my understanding is this would be a child under Segment Store HealthContributor..... Since there is no separate Health REST service for Bookkeeper?

Reply:
That's right. Are you suggesting it be handled differently? I'm not sure I understand your concern.

Reply:
No. I'm not suggesting it to be handled differently. I'm just trying to get clarity on how this is expected to work. Cool. I think this is great